x-pack/platform/plugins/shared/cases/server/cases_analytics/tasks/backfill_task/backfill_task_runner.ts

```diff
@@ -0,0 +1,145 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import type { Logger } from '@kbn/logging';
+import {
+  createTaskRunError,
+  TaskErrorSource,
+  throwRetryableError,
+  throwUnrecoverableError,
+  type ConcreteTaskInstance,
+} from '@kbn/task-manager-plugin/server';
+import type { ElasticsearchClient } from '@kbn/core/server';
+import type { CancellableTask } from '@kbn/task-manager-plugin/server/task';
+import type {
+  IndicesGetMappingResponse,
+  QueryDslQueryContainer,
+} from '@elastic/elasticsearch/lib/api/types';
+import { isRetryableEsClientError } from '../../utils';
+
+interface BackfillTaskRunnerFactoryConstructorParams {
+  taskInstance: ConcreteTaskInstance;
+  getESClient: () => Promise<ElasticsearchClient>;
+  logger: Logger;
+}
+
+export class BackfillTaskRunner implements CancellableTask {
+  private readonly sourceIndex: string;
+  private readonly destIndex: string;
+  private readonly sourceQuery: QueryDslQueryContainer;
+  private readonly getESClient: () => Promise<ElasticsearchClient>;
+  private readonly logger: Logger;
+  private readonly errorSource = TaskErrorSource.FRAMEWORK;
+
+  constructor({ taskInstance, getESClient, logger }: BackfillTaskRunnerFactoryConstructorParams) {
+    this.sourceIndex = taskInstance.params.sourceIndex;
+    this.destIndex = taskInstance.params.destIndex;
+    this.sourceQuery = taskInstance.params.sourceQuery;
+    this.getESClient = getESClient;
+    this.logger = logger;
+  }
+
+  public async run() {
+    const esClient = await this.getESClient();
+    try {
+      await this.waitForDestIndex(esClient);
+      await this.backfillReindex(esClient);
+
+      return {
+        // one time only tasks get deleted so this state is not enough
+        // for the periodic tasks to know the backfill was complete
+        state: {}, // ?
+      };
```

Comment:
In the synchronization PR, I will change this to be the `taskId` returned in `esClient.reindex`.

The synchronization task then does something like:
```
const taskResponse = await esClient.tasks.get({
  task_id: taskId,
  wait_for_completion: false,
});

if (!taskResponse.completed) {
...
}
```

Reply:
We should also save the backfill task's start time. This will help with the first synchronization task.

The synchronization task, being periodic, will not have a value for `last_synched_at` on the first execution. `backfill_started_at` is a good value to look at.